src/gwc/src/main/java/org/geoserver/gwc/GWC.java

```diff
@@ -2058,5 +2066,151 @@ public void testQuotaConfiguration(JDBCConfiguration jdbcConfiguration) throws C
     public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
         this.applicationContext = applicationContext;
     }
+    
+    /**
+     * @return the list of configured blobstores
+     */
+    public List<BlobStoreConfig> getBlobStores() {
+        XMLConfiguration xmlConfig = getXmlConfiguration();
+
+        return new ArrayList<BlobStoreConfig>(xmlConfig.getBlobStores());
+    }
+
+    /**
+     * @return the {@link BlobStoreConfig#isDefault() default} blobstore, or {@code null} if there's
+     *         no default
+     */
+    public BlobStoreConfig getDefaultBlobStore() {
+        XMLConfiguration xmlConfig = getXmlConfiguration();
+
+        for (BlobStoreConfig config : xmlConfig.getBlobStores()) {
+            if (config.isDefault()) {
+                return config;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Convenience method to add a new blob store, calling {@link #setBlobStores} the extra
+     * {@code config}
+     */
+    public void addBlobStore(BlobStoreConfig config) throws ConfigurationException {
+        checkNotNull(config);
+
+        List<BlobStoreConfig> stores = new ArrayList<>(getXmlConfiguration().getBlobStores());
+        if (config.isDefault()) {
+            for (BlobStoreConfig c : stores) {
+                c.setDefault(false);
+            }
+        }
+        stores.add(config);
 
+        setBlobStores(stores);
+    }
+
+    /**
+     * Convenience method to modify a blobstore; calling {@link #setBlobStores(List)} with the
```

Comment:
"Modify a blobstore" is a bit vague.  Maybe "Update the configuration of a blobstore" would be clearer.